src/Microsoft.Data/DataFrame.cs

```diff
@@ -92,5 +113,55 @@ public IList<IList<object>> Tail(int numberOfRows)
         }
         // TODO: Add strongly typed versions of these APIs
         #endregion
+
+        private DataFrame Clone()
+        {
+            List<BaseColumn> newColumns = new List<BaseColumn>(ColumnCount);
+            for (int i = 0; i < ColumnCount; i++)
+            {
+                newColumns.Add(Column(i).Clone());
+            }
+            return new DataFrame(newColumns);
+        }
+
+        public DataFrame Sort(string columnName, bool ascending = true)
+        {
+            BaseColumn column = this[columnName];
+            BaseColumn sortIndices = column.GetAscendingSortIndices();
```

Comment:
we should only call public BaseColumn methods so this works with external derived classes. As it exists today -  this will throw an exception because there is no way for an external column to override an internal method.